pkg/metrics/monotonic_count.go

```diff
@@ -28,16 +39,20 @@ func (mc *MonotonicCount) addSample(sample *MetricSample, timestamp float64) {
 		mc.hasPreviousSample = true
 	}
 
+	mc.flushFirstValue = sample.FlushFirstValue
+
 	// To handle cases where the samples are not monotonically increasing, we always add the difference
 	// between 2 consecutive samples to the value that'll be flushed (if the difference is >0).
 	diff := mc.currentSample - mc.previousSample
-	if mc.hasPreviousSample && diff > 0. {
+	if (mc.hasPreviousSample || mc.flushFirstValue) && diff > 0. {
 		mc.value += diff
+	} else if mc.flushFirstValue {
+		mc.value = mc.currentSample
 	}
 }
 
 func (mc *MonotonicCount) flush(timestamp float64) ([]*Serie, error) {
-	if !mc.sampledSinceLastFlush || !mc.hasPreviousSample {
+	if !mc.sampledSinceLastFlush || !(mc.hasPreviousSample || mc.flushFirstValue) {
```

Comment:
Are you sure that  this `if !mc.sampledSinceLastFlush` is still needed since as far as I understand the sampling isn't needed for the first value? Testing `hasPreviousSample` may be enough? I may be totally wrong.

Reply:
Yes, this is still needed, otherwise if no sample is submitted at all on an existing `MonotonicCount` it would flush a `0` value, whereas in this case we want it to flush no value at all

Reply:
👍  !